[Ryan Cheung] iP

RyanCheungJF opened the pull request:
## Duke Pro

- I
- ~~hate~~ **love**
- Java

1. I
2. love
3. verbosity

```java
public class Main {
    public static void main(String[] args) {
        System.out.println("Hello World!🥴");
    }
}
```
- [ ] survive
- [x] eat
- [ ] sleep

:octocat:

> One man’s crappy software is another man’s full-time job. (Jessica Gaston) ([source](https://www.journaldev.com/240/my-25-favorite-programming-quotes-that-are-funny-too))

If nothing works, just `git push -f <remote> <branch>`

~~Pft who needs stackoverflow~~ _Why_ is stackoverflow **down**??

RyanCheungJF committed:
Implemented Level 1

RyanCheungJF committed:
Implemented Level 2

RyanCheungJF committed:
Implemented Level 3

RyanCheungJF committed:
Implemented Level 4

RyanCheungJF committed:
Implemented Automated Testing

RyanCheungJF committed:
Implemented Level 5

RyanCheungJF committed:
Implemented Level 6

RyanCheungJF committed:
Changed to Linked Lists

RyanCheungJF committed:
Fix styling issues

RyanCheungJF committed:
Implement Level 7 and Refactor

RyanCheungJF committed:
Implement Level 8 with date time functionality

RyanCheungJF committed:
Merge branch 'branch-Level-7'

RyanCheungJF committed:
Fix merge commits from Level-8 branch to master

RyanCheungJF committed:
Refactor functions into respective classes

RyanCheungJF committed:
Package classes respectively

RyanCheungJF committed:
Implement test cases on certain classes

RyanCheungJF committed:
Initialize gradle file in project directory

RyanCheungJF committed:
Add Javadocs to improve readability

RyanCheungJF committed:
Implement find function

RyanCheungJF committed:
Import checkstyle file and configure IDE settings

RyanCheungJF committed:
Merge branch 'branch-A-JavaDoc'

RyanCheungJF committed:
Fix merge conflicts and add Javadocs for new methods

RyanCheungJF committed:
Create config checkstyle files

RyanCheungJF committed:
Fix according to gradle checkstyle

RyanCheungJF committed:
Change variable name to refect coding standard

RyanCheungJF committed:
Add gradle support for unit testing

RyanCheungJF committed:
Initialize basic javafx gui

RyanCheungJF committed:
Add javafx support for gui

RyanCheungJF committed:
Merge branch 'branch-Level-10'

RyanCheungJF committed:
Fix checkstyle errors in latest push

RyanCheungJF committed:
Change styling

RyanCheungJF committed:
Fix GUI issues

kevinkuo0320 commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,131 @@
+     * @return An array of the date and time format, if specified.
+     * @throws DukeException If any of the params do not satisfy conditions for processing.
+     */
+    public static String[] isTaskValid(int index, String input, String taskType) throws DukeException {
```
Good naming for the method

kevinkuo0320 commented on src/main/java/duke/handler/Handlers.java:
```diff
@@ -0,0 +1,256 @@
+/**
+ * Enum class used to handle the different commands used in the application.
+ */
+public enum Handlers {
```
Great application of using enum class!

kevinkuo0320 reviewed (commented):
Overall good job on your ip!

RyanCheungJF committed:
Add assert into functions

Current implementation has no assertions

Add assert into functions

Ensure assumptions of code hold

Prevent errors from programmers

RyanCheungJF committed:
Merge pull request #1 from RyanCheungJF/branch-A-Assertions

Add assert into functions

RyanCheungJF committed:
Refactor code according to code quality

Current implementation has space for refactoring

Improves readability

Add code guards and constants

Prevents use of magic numbers

RyanCheungJF committed:
Merge pull request #2 from RyanCheungJF/branch-A-CodeQuality

Refactor code according to code quality

RyanCheungJF committed:
Add continuous integration

RyanCheungJF committed:
Permission access for yml

RyanCheungJF committed:
Add C-DetectDuplicates

RyanCheungJF committed:
Refactor with respect to codequality

RyanCheungJF committed:
Merge pull request #3 from RyanCheungJF/branch-BCD-Extension

Branch BCD-Extension

RyanCheungJF committed:
Fix syntax for checkstyle

CS2103Lerulatus commented on src/main/java/duke/handler/Handlers.java:
```diff
@@ -0,0 +1,265 @@
+        String[] inputArr = input.split(" ");
+        String cmd = inputArr[0];
+        try {
+            if (cmd.equals(Bye.label)) {
+                return Handlers.byeHandler(list);
+            } else if (cmd.equals(List.label)) {
+                return Handlers.listHandler(list);
+            } else if (cmd.equals(Deadline.label)) {
+                return Handlers.deadlineHandler(list, input, cmd);
+            } else if (cmd.equals(Event.label)) {
+                return Handlers.eventHandler(list, input, cmd);
+            } else if (cmd.equals(Todo.label)) {
+                return Handlers.todoHandler(list, input);
+            } else if (cmd.equals(Mark.label)) {
+                return Handlers.markHandler(list, input);
+            } else if (cmd.equals(Unmark.label)) {
+                return Handlers.unmarkHandler(list, input);
+            } else if (cmd.equals(Find.label)) {
+                return Handlers.findHandler(list, input);
+            } else if (cmd.equals(Delete.label)) {
+                return Handlers.deleteHandler(list, input);
+            } else {
+                throw new IllegalArgumentException("Sorry, this is not a recognized command.\n");
```
——— For PR review ———
Consider looping through the enums instead, for less verbosity and better code readability.
ie. something like:
```
for (Handler cmdType : Handler.values()) {
    if(cmd.equals(cmdType.label))
        return [Handler based on cmdType]
    ...
```
This impl. requires no if-statement addition whenever a new command is added.
Some code restructuring here is required to achieve it but it should be worth it.

Otherwise, good use of the java enum feature I reckon!
Codestyle and indentation here lgtm.

CS2103Lerulatus commented on src/main/java/duke/handler/FileHandler.java:
```diff
@@ -0,0 +1,130 @@
+import java.io.IOException;
+import java.nio.file.Files;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Tasklist;
+import duke.task.Todo;
```
——— For PR review ———
Nice and clear use of imports, even when import `duke.task.*` is tempting to write when you are clearly using all every component inside your `task` package.
Individual imports helps readability.

RyanCheungJF committed:
Update User Guide

RyanCheungJF committed:
Set theme jekyll-theme-slate

RyanCheungJF committed:
Refactor code to SLAP harder

RyanCheungJF committed:
Merge branch 'master' of https://github.com/RyanCheungJF/ip

RyanCheungJF committed:
Change README to make it more comprehensive

RyanCheungJF committed:
Increase redability by refactoring text

RyanCheungJF committed:
Update README

RyanCheungJF committed:
Update README
